openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +115,225 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
+
+    def test_get_custom_pacing_children(self):
+        """
+        _get_custom_pacing_items should return a list of (block item location, field metadata dictionary)
+        where the due dates are set from due_num_weeks
+        """
+        # A subsection with multiple units but no problems. Units should inherit due date.
+        with modulestore().bulk_operations(self.course.id):
+            sequence = ItemFactory(parent=self.course, category='sequential', due_num_weeks=2)
+            vertical1 = ItemFactory(parent=sequence, category='vertical')
+            vertical2 = ItemFactory(parent=sequence, category='vertical')
+            vertical3 = ItemFactory(parent=sequence, category='vertical')
+            expected_dates = [
+                (sequence.location, {'due': timedelta(weeks=2)}),
+                (vertical1.location, {'due': timedelta(weeks=2)}),
+                (vertical2.location, {'due': timedelta(weeks=2)}),
+                (vertical3.location, {'due': timedelta(weeks=2)})
+            ]
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+            # A subsection with multiple units, each of which has a problem.
+            # Problems should also inherit due date.
+            problem1 = ItemFactory(parent=vertical1, category='problem')
+            problem2 = ItemFactory(parent=vertical2, category='problem')
+            expected_dates.extend([
+                (problem1.location, {'due': timedelta(weeks=2)}),
+                (problem2.location, {'due': timedelta(weeks=2)})
+            ])
+            sequence = modulestore().get_item(sequence.location)
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+            # A subsection that has ORA as a problem. ORA should not inherit due date.
+            ItemFactory.create(parent=vertical3, category='openassessment')
+            sequence = modulestore().get_item(sequence.location)
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+            # A subsection that has an ORA problem and a non ORA problem. ORA should
+            # not inherit due date, but non ORA problems should.
+            problem3 = ItemFactory(parent=vertical3, category='problem')
+            expected_dates.append((problem3.location, {'due': timedelta(weeks=2)}))
+            sequence = modulestore().get_item(sequence.location)
+            self.assertCountEqual(_get_custom_pacing_children(sequence, 2), expected_dates)
+
+
+class SelfPacedCustomDueDateTests(SharedModuleStoreTestCase):
+    def setUp(self):
+        SelfPacedRelativeDatesConfig.objects.create(enabled=True)
+
+        # setUpClassAndTestData() already calls setUpClass on SharedModuleStoreTestCase
+        # pylint: disable=super-method-not-called
+        with super().setUpClassAndTestData():
+            self.courses = []
```

Comment:
So I think I would just rewrite this to instead make the course and chapter and just assign to their own variables. You don’t need to go through the get course process
I.e. self.course = CourseFactory
Self.chapter= ItemFactory